crates/router/src/db/cache.rs

```diff
@@ -54,3 +83,22 @@ where
         .change_context(errors::StorageError::KVError)?;
     Ok(data)
 }
+
+pub async fn publish_and_redact<T, F, Fut>(
+    store: &Store,
+    key: &str,
+    fun: F,
+) -> CustomResult<T, errors::StorageError>
+where
+    F: FnOnce() -> Fut + Send,
+    Fut: futures::Future<Output = CustomResult<T, errors::StorageError>> + Send,
+{
+    let data = fun().await?;
+    store
+        .redis_conn()
+        .map_err(Into::<errors::StorageError>::into)?
+        .publish(consts::PUB_SUB_CHANNEL, key)
```

Comment:
where delete from cache is happening?, this is only publishing right?

Reply:
Deleting happens in `on_message` function of pubsub interface